widgets/src/main/java/org/pentaho/gwt/widgets/client/filechooser/FileChooserDialog.java

```diff
@@ -405,17 +405,9 @@ public void setSubmitOnEnter( boolean submitOnEnter ) {
   public boolean onKeyDownPreview( char key, int modifiers ) {
     // Use the popup's key preview hooks to close the dialog when either
     // enter or escape is pressed.
-    switch ( key ) {
-      case KeyboardListener.KEY_ENTER:
-        if ( isSubmitOnEnter() ) {
-          onOk();
-        }
-        break;
-      case KeyboardListener.KEY_ESCAPE:
-        onCancel();
-        break;
+    if ( key == (char) KeyboardListener.KEY_ESCAPE ) {
+      onCancel();
```

Comment:
Please add a deprecation annotation and comment on the "submit on enter" methods and remove the field. The same would apply to the `FileChooser`'s one methods and field.